pkg/httpclient/client.go

```diff
@@ -0,0 +1,70 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+)
+
+// A Client is an HTTP client.
+type Client struct {
+	acsToken string
+	baseURL  string
+	client   *http.Client
+}
+
+// New returns a new HTTP client based on a Config.
+func New(conf config.Config) *Client {
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: conf.TLS().Insecure,
+		RootCAs:            conf.TLS().RootCAs,
+	}
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   conf.Timeout(),
+	}
+
+	return &Client{
+		acsToken: conf.ACSToken(),
+		baseURL:  conf.URL(),
+		client:   client,
+	}
+}
+
+// Get issues a GET to the specified DC/OS cluster path.
+func (c *Client) Get(path string) (resp *http.Response, err error) {
+	req, err := http.NewRequest("GET", c.baseURL+path, nil)
+	if err != nil {
+		return nil, err
+	}
+	return c.Do(req)
+}
+
+// Post issues a POST to the specified DC/OS cluster path.
+func (c *Client) Post(path string, contentType string, body io.Reader) (resp *http.Response, err error) {
+	req, err := http.NewRequest("POST", c.baseURL+path, body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", contentType)
+	return c.Do(req)
+}
+
+// NewRequest returns a new Request given a method, path, and optional body.
+func (c *Client) NewRequest(method, path string, body io.Reader) (req *http.Request, err error) {
+	return http.NewRequest(method, c.baseURL+path, body)
+}
+
+// Do sends an HTTP request and returns an HTTP response, following
+// policy (such as redirects, cookies, auth) as configured on the
+// client. Also sends the authorization header with the ACS token
+// to work with the DC/OS cluster we are linked to.
+func (c *Client) Do(req *http.Request) (resp *http.Response, err error) {
+	req.Header.Set("Authorization", "token="+c.acsToken)
```

Comment:
I'd do this in NewRequest.

Reply:
The reasoning being that this client might also be created with no ACS token yet, and later on one would want to log in and make a request with the returned token, but here it would always be overwritten with an empty one.

Reply:
There should also be an if condition to only send this header if the acs token is not empty.